bin/check-consul-stale-peers.rb

```diff
@@ -0,0 +1,131 @@
+#! /usr/bin/env ruby
+# frozen_string_literal: true
+
+#
+#   check-consul-stale-peers
+#
+# DESCRIPTION:
+#   This plugin checks the raft configuration for stale ("unknown") peers.
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: rest-client
+#
+# USAGE:
+#   Connect to localhost, go critical when there are any stale peers:
+#     ./check-consul-stale-peers
+#
+#   Connect to a remote Consul server over HTTPS:
+#     ./check-consul-stale-peers -s 192.168.42.42 -p 4443 -S https
+#
+#   Go critical when the cluster has two or more stale peers:
+#     ./check-consul-stale-peers -W 1 -C 2
+#
+# NOTES:
+#
+# LICENSE:
+#   Copyright 2015 Sonian, Inc. and contributors. <support@sensuapp.org>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'json'
+require 'rest-client'
+require 'sensu-plugin/check/cli'
+
+#
+# Consul stale peers
+#
+class ConsulStalePeers < Sensu::Plugin::Check::CLI
+  option :server,
+         description: 'consul server',
+         short: '-s SERVER',
+         long: '--server SERVER',
+         default: '127.0.0.1'
+
+  option :port,
+         description: 'consul http port',
+         short: '-p PORT',
+         long: '--port PORT',
+         proc: proc(&:to_i),
+         default: '8500'
+
+  option :scheme,
+         description: 'consul listener scheme',
+         short: '-S SCHEME',
+         long: '--scheme SCHEME',
+         default: 'http'
+
+  option :insecure,
+         description: 'set this flag to disable SSL verification',
+         short: '-k',
+         long: '--insecure',
+         boolean: true,
+         default: false
+
+  option :capath,
+         description: 'absolute path to an alternative CA file',
+         short: '-c CAPATH',
+         long: '--capath CAPATH'
+
+  option :timeout,
+         description: 'connection will time out after this many seconds',
+         short: '-t TIMEOUT_IN_SECONDS',
+         long: '--timeout TIMEOUT_IN_SECONDS',
+         proc: proc(&:to_i),
+         default: 5
+
+  option :warning,
+         description: 'warn when there are this many stale peers',
+         short: '-W NUMBER_OF_PEERS',
+         long: '--warning NUMBER_OF_PEERS',
+         proc: proc(&:to_i),
+         default: 1
+
+  option :critical,
+         description: 'go critical when there are this many stale peers',
+         short: '-C NUMBER_OF_PEERS',
+         long: '--critical NUMBER_OF_PEERS',
+         proc: proc(&:to_i),
+         default: 1
+
+  #
+  # Fetch and return the parsed JSON data from a specified Consul API endpoint.
+  #
+  def consul_get(endpoint)
```

Comment:
Would it be worth doing this with the other plugins that use the same code as well? Or would that be considered a backwards-incompatible change since the plugins would no longer be entirely self-sufficient?

Reply:
we generally prefer to keep each plugin (the gem/pip) self contained. You could add something similar in `sensu-plugin` for anything shared as pretty much all of them depend on it we would just need to bump the dep. I'd for now just stick it in a lib in the gem and call it good.